drivers/bbram/bbram_shell.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/util.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/drivers/bbram.h>
+
+/* Buffer is only needed for bytes that follow command, device and address. */
+#define BUF_ARRAY_CNT (CONFIG_SHELL_ARGC_MAX - 3)
+
+static int parse_ul(const char *str, unsigned long *result)
+{
+	char *end;
+	unsigned long val;
+
+	val = strtoul(str, &end, 0);
+
+	if (*str == '\0' || *end != '\0') {
+		return -EINVAL;
+	}
+
+	*result = val;
+	return 0;
+}
+
+static int parse_u32(const char *str, uint32_t *result)
+{
+	unsigned long val;
+
+	if (parse_ul(str, &val) || val > 0xffffffff) {
+		return -EINVAL;
+	}
+	*result = (uint32_t)val;
+	return 0;
+}
+
+static int parse_u8(const char *str, uint8_t *result)
+{
+	unsigned long val;
+
+	if (parse_ul(str, &val) || val > 0xff) {
+		return -EINVAL;
+	}
+	*result = (uint8_t)val;
+	return 0;
+}
```

Comment:
Why do we need these? `stdlib` includes `strtoul`, can we use that?

Reply:
Actually, `strtoul` is used in line 25, these functions are a wrapper for `strtoul` call with error handling.
Function `parse_ul` checks `end` variable, so we know if there was a problem with parsing.
Functions `parse_u32` and `parse_u8` check if number is in expected range.

Reply:
Added.